### ISSUE TITLE
IPRO check-fails on some connection errors

### ISSUE THREAD

**Issue by grayfolk** [pagespeed/ngx_pagespeed#1096]
When i turn pagespeed on then have aborted connection and this error.log message:
```
2016/01/19 20:11:55 [alert] 26489#0: [ngx_pagespeed 1.10.33.2-7600] [0119/201155:FATAL:ngx_base_fetch.cc(144)] Check failed: r->count > 0. r->count: 0
nginx: worker process() [0x813edcb]
nginx: worker process() [0x80e16f0]
nginx: worker process() [0x813b9ac]
nginx: worker process() [0x80e2638]
nginx: worker process() [0x80e3102]
nginx: worker process() [0x80e3173]
nginx: worker process() [0x806b8bc]
nginx: worker process() [0x806b48a]
nginx: worker process() [0x80711f2]
nginx: worker process() [0x806fb0f]
nginx: worker process() [0x8072282]
nginx: worker process() [0x8052bd2]
/lib/libc.so.6(__libc_start_main+0xe6) [0x795d36]
nginx: worker process() [0x8052f61]
2016/01/19 20:11:55 [alert] 26475#0: worker process 26489 exited on signal 6
```
Pagespeed config: (inside server{} section)
```
pagespeed On;
pagespeed RewriteLevel CoreFilters;

# HTTPS Support
pagespeed FetchHttps enable,allow_self_signed;

pagespeed EnableFilters lazyload_images,collapse_whitespace,insert_dns_prefetch,dedup_inlined_images,defer_javascript,pedantic,trim_urls,sprite_images,extend_cache_pdfs,remove_comments,resize_mobile_images,inline_preview_images,insert_image_dimensions,convert_to_webp_lossless,local_storage_cache,inline_google_font_css,prioritize_critical_css,rewrite_style_attributes,move_css_to_head,move_css_above_scripts,outline_javascript,outline_css,combine_heads;

pagespeed FileCachePath /var/ngx_pagespeed_cache;

location ~ "\.pagespeed\.([a-z]\.)?[a-z]{2}\.[^.]{10}\.[^.]+" { add_header "" ""; }
location ~ "^/pagespeed_static/" { }
location ~ "^/ngx_pagespeed_beacon$" { }
```
nginx 1.8.0, CentOS 6.7

When i change to pagespeed Off; - nginx working.
```
# nginx -t
nginx: the configuration file /etc/nginx/nginx.conf syntax is ok
nginx: configuration file /etc/nginx/nginx.conf test is successful
```

**Comment by oschaaf** [comment]
- Are you running http/2?
- Would you be able to  PM your full configuration and site url plus the contents of nginx -V?
- Would you be able to run a debug build, configure nginx to log debug level messages, and post the resulting contents of error.log (or PM) the contents of error.log ?

**Comment by jeffkaufman** [comment]
Fixed by b88e067c6d8173112d586fcdfe8d664c1226a00d
